### ISSUE TITLE
KITTI training performance

### ISSUE THREAD

**Issue by zxc-2015** [packyan/PyTorch-YOLOv3-kitti#6]
Hi packyan,
The KITTI val performance of using the pre-trained model is very nice, which is:
Average Precisions:
+ Class '0' - AP: 0.8874476379466757
+ Class '1' - AP: 0.8776504929104587
+ Class '2' - AP: 0.9094174650326788
+ Class '3' - AP: 0.7935572902584845
+ Class '4' - AP: 0.6938401233821029
+ Class '5' - AP: 0.7484522134591276
+ Class '6' - AP: 0.9765143493123802
+ Class '7' - AP: 0.6567834290009646
mAP: 0.8179578751628592

However, when doing the training with the Imagenet pretrained-model, I am only able to get the following results, which is:
Average Precisions:
+ Class '0' - AP: 0.5895481840925929
+ Class '1' - AP: 0.056750284687503164
+ Class '2' - AP: 0.05041402945683988
+ Class '3' - AP: 0.25446338964588194
+ Class '4' - AP: 0.0
+ Class '5' - AP: 0.042098299062717656
+ Class '6' - AP: 0.0
+ Class '7' - AP: 0.0
mAP: 0.12415927336819195

I'm wondering what thing do I miss or configure? I use the same label-transforming pre-processing code as in the instruction and training without changing the python arguments. Is anything wrong I did or any arguments need to be changed?

Thanks a lot!

**Comment by packyan** [comment]
Imagenet pretrained-model is just for darknet layer, the latter layers you should train by kitti dataset.

**Comment by zhaiyi000** [comment]
Average Precisions:
Car 				+ Class '0' - AP: 0.8874329809457615   
Van 				+ Class '1' - AP: 0.8776504929104587   
Truck 				+ Class '2' - AP: 0.9094174650326788   
Pedestrian 			+ Class '3' - AP: 0.7935575396672143   
Person_sitting 		+ Class '4' - AP: 0.6938401233821029   
Cyclist 			+ Class '5' - AP: 0.7484521975616071   
Tram 				+ Class '6' - AP: 0.9765143493123802   
Misc 				+ Class '7' - AP: 0.656781273677661   
mAP: 0.817955802811233


Average Precisions:
Car					+ Class '0' - AP: 0.5925623770514821
Van					+ Class '1' - AP: 0.06155695621575082
Truck				+ Class '2' - AP: 0.046829029317335555
Pedestrian			+ Class '3' - AP: 0.21473984288447187
Person_sitting		+ Class '4' - AP: 0.0
Cyclist				+ Class '5' - AP: 0.04053911530526016
Tram				+ Class '6' - AP: 0.0
Misc				+ Class '7' - AP: 0.0
mAP: 0.11952841509678755


Hi zxc-2015, I encountered the same problem as you.
I divided the kitti 7481 training images into 3712(train)+3769(val)
Did you solve this problem？

I am trying the whole training set to train the model, and will see the result after 100 epoch

**Comment by wewan** [comment]
> Me too, did you solve it?
